src/library/scala/collection/generic/Subtractable.scala

```diff
@@ -59,5 +61,10 @@ trait Subtractable[A, +Repr <: Subtractable[A, Repr]] { self =>
    *  @return a new $coll that contains all elements of the current $coll
    *  except one less occurrence of each of the elements of `elems`.
    */
-  def --(xs: GenTraversableOnce[A]): Repr = (repr /: xs.seq) (_ - _)
+  def --(xs: GenTraversableOnce[A]): Repr = this match {
```

Comment:
This will impact all immutable maps and sets. OTOH, the default implementation is rather inefficient. The extra cost for the type check shouldn't be too bad.